### DISCUSSION TITLE
How to get Headers in $fetch?

### DISCUSSION THREAD

**Discussion by k0mar12** [nuxt/nuxt#15943] [Questions] [answered]
Im using laravel on backend and sanctum for authorization.

I created plugin to execute him on server side, fetch all needed cookies and pass them on client side, but i cant get headers according to **ohmyfetch** documentation all headers, status code etc shoud be in **response**. There is nothink

```
export default defineNuxtPlugin(async nuxtApp => {
    const response = await $fetch('http://backend.local/sanctum/csrf-cookie');

    console.log(response);
});
```

So, how i can get headers, cookies, status code while using $fetch ? 

response.headers = undefined
response.statusText = undefined 
response.ok = undefined 
response.error = undefined

**Comment by danielroe** [comment, marked as answer]
You can use `$fetch.raw()` instead of `$fetch()` if you need access to the underlying response object.

**Reply by k0mar12** [reply]
yes, Your right, i need read docs better, thank You :)

**Reply by Bhagath96** [reply]
after i choosing $fetch.raw() the response.headers is {} empty object.   response.headers.get('record-count') that is null. How can i get the 'record-count' value in the respons header.

**Reply by idesignzone** [reply]
how to refresh data when using `$fetch.raw()` ?

**Reply by valentinhttr** [reply]
Hi @Bhagath96 & @nikdejan
It is not empty, the instance of the Headers class returned by the fetch function is an iterable and some of the data, such as Headers are not directly visible from the console.

So to see them, you have to loop trough them like this:
```ts
for(let entry of res.headers.entries()) {
    console.log(entry);
}
```

**Reply by Virock** [reply]
Hi. I understand that $fetch (when in useAsyncData and running on the server) does not make a HTTP call. It just runs the endpoint like a method and returns the result. Is this  the same for $fetch.raw()? I checked online but can't find and documentation for $fetch.raw()
